canister/src/api/set_config.rs

```diff
@@ -2,8 +2,33 @@ use ic_btc_interface::SetConfigRequest;
 use std::convert::TryInto;
 
 pub async fn set_config(request: SetConfigRequest) {
-    verify_caller().await;
-    set_config_no_verification(request);
+    if is_watchdog_caller() {
```

Comment:
I think we should include some test cases here:

1. Watchdog calling `set_config` with `api_access` actually updates `api_access`.
2. Watchdog calling `set_config` with changes other than `api_access` are ignored.

You can actually accomplish with unit tests - no need to go all the way and do e2e tests.

Reply:
Done, added mocked caller & controllers, PTAL.